en/ros2/px4_ros2_interface_lib.md

```diff
@@ -0,0 +1,365 @@
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Got RegisterExtComponentReply
+   [DEBUG] [example_mode_manual]: Arming check request (id=1, only printed once)
+   ```
+
+8. On the PX4 shell, you can check that PX4 registered the new mode:
+
+   ```sh
+   commander status
+   ```
+
+   The output should contain:
+
+   ```{5}
+   INFO  [commander] Disarmed
+   INFO  [commander] navigation mode: Position
+   INFO  [commander] user intended navigation mode: Position
+   INFO  [commander] in failsafe: no
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+   ```
+
+9. At this point you should be able to see the mode in QGroundControl as well:
+
+   ![QGC Modes](../../assets/middleware/ros2/px4_ros2_interface_lib/qgc_modes.png)
+
+10. Select the mode, make sure you have a manual control source (physical or virtual joystick), and arm the vehicle.
+    The mode will then activate, and it should print the following output:
+
+    ```sh
+    [DEBUG] [example_mode_manual]: Mode 'My Manual Mode' activated
+    ```
+
+11. Now you are ready to create your own mode.
+
+## How to Use the Library
+
+The following sections describe specific functionality provided by this library.
+In addition, any other PX4 topic can be subscribed or published.
+
+### Mode Class Definition
+
+This section steps through an example of how to create a class for a custom mode.
+
+For a complete application, check out the [examples in the `Auterion/px4-ros2-interface-lib` repository](https://github.com/Auterion/px4-ros2-interface-lib/tree/main/examples/cpp), such as [examples/cpp/modes/manual](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/examples/cpp/modes/manual/include/mode.hpp).
+
+```cpp{1,5,7-9,24-31}
+class MyMode : public px4_ros2::ModeBase // [1]
+{
+public:
+  explicit MyMode(rclcpp::Node & node)
+  : ModeBase(node, Settings{"My Mode"}) // [2]
+  {
+    // [3]
+    _manual_control_input = std::make_shared<px4_ros2::ManualControlInput>(*this);
+    _rates_setpoint = std::make_shared<px4_ros2::RatesSetpointType>(*this);
+  }
+
+  void onActivate() override
+  {
+    // Called whenever our mode gets selected
+  }
+
+  void onDeactivate() override
+  {
+    // Called when our mode gets deactivated
+  }
+
+  void updateSetpoint(const rclcpp::Duration & dt) override
+  {
+    // [4]
+    const Eigen::Vector3f thrust_sp{0.F, 0.F, -_manual_control_input->throttle()};
+    const Eigen::Vector3f rates_sp{
+      _manual_control_input->roll() * 150.F * M_PI / 180.F,
+      -_manual_control_input->pitch() * 150.F * M_PI / 180.F,
+      _manual_control_input->yaw() * 100.F * M_PI / 180.F
+    };
+    _rates_setpoint->update(rates_sp, thrust_sp);
+  }
+
+private:
+  std::shared_ptr<px4_ros2::ManualControlInput> _manual_control_input;
+  std::shared_ptr<px4_ros2::RatesSetpointType> _rates_setpoint;
+};
+```
+
+- `[1]`: First we create a class that inherits from `px4_ros2::ModeBase`
```

Comment:
Do you think we could create a "reference" of the public API - just names and a description for the main classes at the end. Starting from a reference tells you what documentation is available/missing.

Reply:
Yes we could use doxygen

Reply:
I am going to assume you have this on your list :-)